### PR TITLE
PD-487: Update TrueNAS Security Best Practices

### PR DESCRIPTION
**Pull request by Mrt134**
This adds a general best practices section in front of the specific service hardening content and reweights headers for better navigation.
Thanks for contributing to TrueNAS documentation! By opening a Pull Request, you're acknowledging that your changes will be distributed under the [Creative Commons 4.0](https://creativecommons.org/licenses/by-nc-sa/4.0/) license.

### CODE REVIEW

File: content/Solutions/Optimizations/Security.md [hunk 1]
```diff
@@ -1,20 +1,36 @@
 ---
 title: "Security Recommendations"
-description: "Best practices when tuning TrueNAS to minimize security vulnerabilities."
+description: "Best practices to minimize security vulnerabilities in TrueNAS."
```
Comment (kmoore134) [inline content/Solutions/Optimizations/Security.md:3]: "Best practices to operate TrueNAS in a secure manner"

File: content/Solutions/Optimizations/Security.md [hunk 2]
```diff
@@ -1,20 +1,36 @@
...
  - scalessh
+keywords:
+ - TrueNAS Security
 ---
 
 {{< toc >}}
 
-When using services on TrueNAS, especially services that allow outside connections, there are some best practices to follow to ensure your system is safe and secure.
-Several different system services are disscused in this article.
+Follow these best practices to minimize security vulnerabilities in TrueNAS.
```
Comment (kmoore134) [inline content/Solutions/Optimizations/Security.md:16]: "Minimize security vulnerabilities" doesn't sound great. Should focus on the positive angle, about best practices to operate TrueNAS in a secure manner.

File: content/Solutions/Optimizations/Security.md [hunk 3]
```diff
@@ -1,20 +1,36 @@
...
 
-## iSCSI
+## General Recommendations
+
+1. Specially modifying the base TrueNAS firmware image is unsupported and can create security issues.
+2. Keep TrueNAS up to date with the most recent maintenance releases of the installed major version.
```
Comment (kmoore134) [inline content/Solutions/Optimizations/Security.md:21]: "Keep TrueNAS up to date with the most recently updates for your supported version"

File: content/Solutions/Optimizations/Security.md [hunk 4]
```diff
@@ -1,20 +1,36 @@
...
+3. Upgrade to new major releases in a timely manner consistent with the deployment use case.
+4. Restrict new TrueNAS user accounts to the most minimal set of permissions and access possible.
+5. Never grant TrueNAS user accounts (local or Directory Services added accounts) access to SSH or login Shells unless that account is explicitly trusted.
```
Comment (kmoore134) [inline content/Solutions/Optimizations/Security.md:24]: "to SSH or console shells"

**Review by kmoore134 — CHANGES_REQUESTED** [review]

**Review by kmoore134 — DISMISSED** [review]

File: content/Solutions/Optimizations/Security.md [hunk 5]
```diff
@@ -1,20 +1,36 @@
...
 
 {{< toc >}}
 
-When using services on TrueNAS, especially services that allow outside connections, there are some best practices to follow to ensure your system is safe and secure.
-Several different system services are disscused in this article.
+Follow these best practices to administrate TrueNAS securely.
 
-## iSCSI
+## General Recommendations
+
```
Comment (micjohnson777) [inline content/Solutions/Optimizations/Security.md:19]: Is this a step procedure or a list of recommendations? Parallel construction is off on a few points.

File: content/Solutions/Optimizations/Security.md [hunk 6]
```diff
@@ -1,20 +1,36 @@
...
+8. Restrict the TrueNAS web, IPMI, and any other management interfaces to private subnets away from untrusted users.
+9. Review any plugin, App, or Virtual Machine (VM) deployment scenario for additional security exposure or vulnerabilities.
+    iXsystems cannot resolve security vulnerabilities introduced from within user-deployed virtualized environments.
+
+## Hardening Specific System Services
+
+### iSCSI
 
 Follow the iSCSI creation wizard unless a specific configuration is required.
 To create an iSCSI share, go to **Sharing > Block Shares (iSCSI)** and click *WIZARD*.
```
Comment (micjohnson777) [inline content/Solutions/Optimizations/Security.md:36]: line 41 the option to add should be in bold not italics.
Line 44 Initiators and Authorized Networks should be bold not italics.

File: content/Solutions/Optimizations/Security.md [hunk 7]
```diff
@@ -29,7 +45,7 @@ Entering a list of *Initiators* and *Authorized Networks* is also recommended.
 This allows defining which systems or networks can connect to the extent.
 When these options are empty, all initiators and all networks are allowed to connect to the extent.
 
-## NFS
+### NFS
 
 Network File System (NFS) is a sharing protocol that allows outside users to connect and view or modify shared data.
```
Comment (micjohnson777) [inline content/Solutions/Optimizations/Security.md:51]: Line 59 Authorized Hosts and IP addresses should be bold not italics.

File: content/Solutions/Optimizations/Security.md [hunk 8]
```diff
@@ -44,7 +60,7 @@ Leaving the **Authorized Networks** or **Authorized Hosts and IP addresses* list
 To define which systems can connect to the share, click the **Advanced Options** and enter all networks, hosts, and IP addresses to have share access.
 All other systems are denied access.
 
-## SMB
+### SMB
 
 Using Server Message Block (SMB) to share data is a very common situation for TrueNAS users.
 However, it allows outside connections to the system and must be properly use to avoid security concerns.
```
Comment (micjohnson777) [inline content/Solutions/Optimizations/Security.md:66]: incorrect use of italics in line 74, 77, 80, 83, and 85, all the UI elements should be bold

**Review by micjohnson777 — DISMISSED** [review]
Need to fix style issues in sections mentioned in comments above the section I could not add the comments to.

File: content/Solutions/Optimizations/Security.md [hunk 9]
```diff
@@ -1,20 +1,36 @@
...
 {{< toc >}}
 
-When using services on TrueNAS, especially services that allow outside connections, there are some best practices to follow to ensure your system is safe and secure.
-Several different system services are disscused in this article.
+Follow these best practices to administrate TrueNAS securely.
 
-## iSCSI
+## General Recommendations
+
+* Specially modifying the base TrueNAS firmware image is unsupported and can create security issues.
```
Comment (micjohnson777) [inline content/Solutions/Optimizations/Security.md:20]: Parallel construction issue, maybe this?
```suggestion
* Modifying  the base TrueNAS firmware image is unsupported and can create security issues.
```

File: content/Solutions/Optimizations/Security.md [hunk 10]
```diff
@@ -1,20 +1,36 @@
...
+* Keep TrueNAS up to date with the most recent updates for your supported version.
+* Upgrade to new major releases in a timely manner consistent with the deployment use case.
+* Use complex passwords and Two-Factor Authentication (2FA) for all TrueNAS Root and Administrator accounts.
+* Restrict new TrueNAS user accounts to the most minimal set of permissions and access possible.
+* Never grant TrueNAS user accounts (local or Directory Services added accounts) access to SSH or console shells unless that account is explicitly trusted.
```
Comment (micjohnson777) [inline content/Solutions/Optimizations/Security.md:25]: Parallel construction issue, maybe this?
```suggestion
* Grant TrueNAS user accounts (local or Directory Services added accounts) access to SSH or console shells only if  that account is explicitly trusted.
```

**Review by micjohnson777 — CHANGES_REQUESTED** [review]
The bullet list had two points that were not observing parallel construction. Made suggestions for a way to change the two points.

**Review by micjohnson777 — APPROVED** [review]

**Pull request merged**